unstoppable-ios-app/domains-manager-ios/Modules/Home/SendCryptoAsset/CryptoSender/CryptoSender+Entities.swift

```diff
@@ -91,16 +91,14 @@ struct EstimatedGasPrices {
 }
 
 struct ChainSpec {
-    let blockchainType: BlockchainType
-    let env: UnsConfigManager.BlockchainEnvironment
+    let chain: BlockchainType.Chain
     
     init(blockchainType: BlockchainType, env: UnsConfigManager.BlockchainEnvironment = .mainnet) {
-        self.blockchainType = blockchainType
-        self.env = env
+        self.chain = blockchainType.resolveChain(env: env)
     }
     
     var id: Int {
-        self.blockchainType.supportedChainId(env: self.env)
+        self.chain.id
     }
 }
```

Comment:
Do we need this ChainSpec wrapper?

Reply:
I would say 70% we need this. From the point of view of the Client it is convenient to say:

I want `Ethereum` but `testnet`

rather than search for appropriate enum. Plus I wouldn't break the protocol for CryptoSender because of this.